src/apps/filters/h3.c

```diff
@@ -0,0 +1,144 @@
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&cellToLatLngArg, &helpArg, &cellArg};
+    if (parseArgs(argc, argv, 3, args, &helpArg,
+                  "Convert an H3 cell to a latitude/longitude coordinate")) {
+        return helpArg.found;
+    }
+    LatLng ll;
+    H3_EXPORT(cellToLatLng)(cell, &ll);
+    printf("%.10lf, %.10lf\n", H3_EXPORT(radsToDegs)(ll.lat),
+           H3_EXPORT(radsToDegs)(ll.lng));
+    return true;
+}
+
+bool latLngToCellCmd(int argc, char *argv[]) {
+    int res = 0;
+    double lat = 0;
+    double lng = 0;
+
+    Arg latLngToCellArg = {
+        .names = {"latLngToCell"},
+        .required = true,
+        .helpText =
+            "Convert degrees latitude/longitude coordinate to an H3 cell.",
+    };
+    Arg helpArg = ARG_HELP;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText = "Resolution, 0-15 inclusive."};
+    Arg latArg = {.names = {"--lat", "--latitude"},
+                  .required = true,
+                  .scanFormat = "%lf",
+                  .valueName = "lat",
+                  .value = &lat,
+                  .helpText = "Latitude in degrees."};
+    Arg lngArg = {.names = {"--lng", "--longitude"},
+                  .required = true,
+                  .scanFormat = "%lf",
+                  .valueName = "lng",
+                  .value = &lng,
+                  .helpText = "Longitude in degrees."};
+
+    Arg *args[] = {&latLngToCellArg, &helpArg, &resArg, &latArg, &lngArg};
+    if (parseArgs(
+            argc, argv, 5, args, &helpArg,
+            "Convert degrees latitude/longitude coordinate to an H3 cell.")) {
+        return helpArg.found;
+    }
+    LatLng ll = {.lat = H3_EXPORT(degsToRads)(lat),
+                 .lng = H3_EXPORT(degsToRads)(lng)};
+
+    H3Index c;
+    H3Error e = H3_EXPORT(latLngToCell)(&ll, res, &c);
+
+    if (e == E_SUCCESS) {
+        h3Println(c);
+    } else {
+        h3Println(H3_NULL);
+    }
+    return true;
+}
+
+bool generalHelp(int argc, char *argv[]) {
+    Arg helpArg = ARG_HELP;
+    Arg cellToLatLngArg = {
+        .names = {"cellToLatLng"},
+        .helpText = "Convert an H3 cell to a latitude/longitude coordinate",
+    };
+    Arg latLngToCellArg = {
+        .names = {"latLngToCell"},
+        .helpText =
+            "Convert degrees latitude/longitude coordinate to an H3 cell.",
+    };
+    Arg *args[] = {&helpArg, &cellToLatLngArg, &latLngToCellArg};
+    const char *helpText =
+        "Please use one of the subcommands listed to perform an H3 "
+        "calculation. Use h3 <SUBCOMMAND> --help for details on the usage of "
+        "any subcommand.";
+    return parseArgs(argc, argv, 3, args, &helpArg, helpText);
+}
+
+int main(int argc, char *argv[]) {
+    if (has("cellToLatLng", 1, argv) && cellToLatLngCmd(argc, argv)) {
```

Comment:
Possible out of range read if argc = 1. (Calls `strcasecmp` with `argv[1] = 0`)?

Reply:
This is true. I could have a simple `if (argc <= 1)` block that prints help and exits immediately to avoid this.